src/hotspot/os/linux/memMapPrinter_linux.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2023, Red Hat, Inc. and/or its affiliates.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+
+#include "runtime/os.hpp"
+#include "services/memMapPrinter.hpp"
+#include "utilities/globalDefinitions.hpp"
+
+struct proc_maps_info_t {
+  unsigned long long from = 0;
+  unsigned long long to = 0;
+  char prot[20 + 1];
+  char offset[20 + 1];
+  char dev[20 + 1];
+  char inode[20 + 1];
+  char filename[1024 + 1];
```

Comment:
That is surprisingly tricky since there is no way to specify variable sizes with scanf, unlike printf. 

One could rely on PATH_MAX being a pure numerical ("4096") and assemble the format string with preprocessor magic, but that falls apart if PATH_MAX is e.g. a constant. 

One could build up the format string dynamically at runtime, but that will trigger warnings in code linters and possibly when building since the format string is supposed to be a literal.

Also, the PATH_MAX size is pretty arbitrary. Paths can get infinitely long with nested links. So there is no safe limit anyway. May just as well keep 1024 then, since that should be enough for 99% of cases.